3party/opening_hours/opening_hours.cpp

```diff
@@ -0,0 +1,1384 @@
+}
+
+std::ostream & operator<<(std::ostream & ost, Weekday const wday)
+{
+  switch (wday)
+  {
+    case Weekday::Sunday:
+      ost << "Su";
+      break;
+    case Weekday::Monday:
+      ost << "Mo";
+      break;
+    case Weekday::Tuesday:
+      ost << "Tu";
+      break;
+    case Weekday::Wednesday:
+      ost << "We";
+      break;
+    case Weekday::Thursday:
+      ost << "Th";
+      break;
+    case Weekday::Friday:
+      ost << "Fr";
+      break;
+    case Weekday::Saturday:
+      ost << "Sa";
+      break;
+    case Weekday::None:
+      ost << "not-a-day";
+  }
+  return ost;
+}
+
+std::ostream & operator<<(std::ostream & ost, WeekdayRange const & range)
+{
+  ost << range.GetStart();
+  if (range.HasEnd())
+    ost << '-' << range.GetEnd();
+  else
+  {
+    if (range.HasNth())
+    {
+      ost << '[';
+      PrintVector(ost, range.GetNths(), ",");
+      ost << ']';
+    }
+    PrintOffset(ost, range.GetOffset(), true);
+  }
+  return ost;
+}
+
+std::ostream & operator<<(std::ostream & ost, TWeekdayRanges const & ranges)
+{
+  PrintVector(ost, ranges);
+  return ost;
+}
+
+
+bool Holiday::IsPlural() const
+{
+  return m_plural;
+}
+
+void Holiday::SetPlural(bool const plural)
+{
+  m_plural = plural;
+}
+
+int32_t Holiday::GetOffset() const
+{
+  return m_offset;
+}
+
+void Holiday::SetOffset(int32_t const offset)
+{
+  m_offset = offset;
+}
+
+std::ostream & operator<<(std::ostream & ost, Holiday const & holiday)
+{
+  if (holiday.IsPlural())
+    ost << "PH";
+  else
+  {
+    ost << "SH";
+    PrintOffset(ost, holiday.GetOffset(), true);
+  }
+  return ost;
+}
+
+std::ostream & operator<<(std::ostream & ost, THolidays const & holidays)
+{
+  PrintVector(ost, holidays);
+  return ost;
+}
+
+
+bool Weekdays::IsEmpty() const
+{
+  return GetWeekdayRanges().empty() && GetHolidays().empty();
+}
+
+bool Weekdays::HasWeekday() const
+{
+  return !GetWeekdayRanges().empty();
+}
+
+bool Weekdays::HasHolidays() const
+{
+  return !GetHolidays().empty();
+}
+
+TWeekdayRanges const & Weekdays::GetWeekdayRanges() const
+{
+  return m_weekdayRanges;
+}
+
+THolidays const & Weekdays::GetHolidays() const
+{
+  return m_holidays;
+}
+
+void Weekdays::SetWeekdayRanges(TWeekdayRanges const ranges)
+{
+  m_weekdayRanges = ranges;
+}
+
+void Weekdays::SetHolidays(THolidays const & holidays)
+{
+  m_holidays = holidays;
+}
+
+void Weekdays::AddWeekdayRange(WeekdayRange const range)
+{
+  m_weekdayRanges.push_back(range);
+}
+
+void Weekdays::AddHoliday(Holiday const & holiday)
+{
+  m_holidays.push_back(holiday);
+}
+
+std::ostream & operator<<(std::ostream & ost, Weekdays const & weekday)
+{
+  ost << weekday.GetHolidays();
+  if (weekday.HasWeekday() && weekday.HasHolidays())
+    ost << ", ";
+  ost << weekday.GetWeekdayRanges();
+  return ost;
+}
+
+
+bool DateOffset::IsEmpty() const
+{
+  return !HasOffset() && !HasWDayOffset();
+}
+
+bool DateOffset::HasWDayOffset() const
+{
+  return m_wday_offset != Weekday::None;
```

Comment:
We don't use unix_hacker_style for class fields. Please, use m_camelCase instead.